pkg/mutation/path/tester/tester.go

```diff
@@ -71,16 +78,40 @@ func ValidatePathTests(location *parser.Path, pathTests []Test) error {
 
 // New creates a new Tester object
 func New(tests []Test) (*Tester, error) {
+	paths := make(map[int]*parser.Path)
 	idx := &Tester{
 		tests: make(map[int]Condition),
 	}
+
+	// Read in all tests before checking for conflicts.
+	idxLowestMustNot := math.MaxInt64
+	idxHighestMust := 0
+	for _, test := range tests {
+		i := len(test.SubPath.Nodes) - 1
```

Comment:
Would this cause issues with siblings?
I'm not sure if there are other mitigating checks, but for example adding this test case to `TestConflictingEntries` fails:

```
{
	name: "siblings should not conflict",
	ts: []Test{
		{
			SubPath:   mustParse("spec.some.thing"),
			Condition: MustExist,
		},
		{
			SubPath:   mustParse("spec.some.other-thing"),
			Condition: MustNotExist,
		},
	},
	wantErr: nil,
},
```

Reply:
It is invalid to create a Tester which tests paths where one path is not an ancestor of the other, so this is working as intended.